packages/api-contract/src/base/Contract.ts

```diff
@@ -141,8 +120,10 @@ export class Contract<ApiType extends ApiTypes> extends Base<ApiType> {
           origin,
           this.address,
           value,
-          // the runtime interface still used u64 inputs
-          this.#getGas(gasLimit, true).v1Weight,
+          this._isOldWeight
```

Comment:
Sorry I don't understand your means, do you mean I need to revert this part change?

Reply:
Yes. It should rather be adapted to al check for the new stuff. You can do this by detecting which version is used, i.e something like -

```js
...
value,
this.api.tx.contracts.call.meta.version === 1
  ? this.#getGas(gasLimit, true).v1Weight
  : this.#getGas(gasLimit, true).v2Weight
...
```

This is important since the getGas also allows the developer to pass a number - since we don't like breaking compat, we have not changed the signatures.

Reply:
Changed as your suggest

Reply:
```
packages/api-contract/src/base/Contract.ts:144:43 - error TS2339: Property 'version' does not exist on type 'FunctionMetadataLatest'.

144           this.api.tx.contracts.call.meta.version === 1
```

I'll dig

Reply:
seems `FunctionMetadataLatest` doesn't have `version` field?

Reply:
@jacogr I still don't got this one